src/Utils/LogRanker.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+ */
+import * as CLM from '@conversationlearner/models'
+import { TreeUtils } from './TreeUtils'
+
+export interface LogScore {
+    logDialogId: string
+    knownPath?: boolean
+    validEndAction?: boolean
+    numberUnknownUtterances?: number
+    numberLowConfidence?: number,
+    score: number
+}
+
+export class LogRanker {
+
+    private entities: CLM.EntityBase[]
+    private logDialogs: CLM.LogDialog[]
+    private trainDialogs: CLM.TrainDialog[]
+    private logScores: LogScore[]
+
+    constructor(logDialogs: CLM.LogDialog[], trainDialogs: CLM.TrainDialog[], entities: CLM.EntityBase[]) {
+        this.entities = entities
+        this.logDialogs = logDialogs
+        this.trainDialogs = trainDialogs
```

Comment:
Why are these member variables rather than parameters to analyze() ?

Reply:
Switched it to just being a util function rather than a class